Acceleratio.XCellKit/SpredsheetChart.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using DocumentFormat.OpenXml;
+using DocumentFormat.OpenXml.Drawing.Charts;
+using DocumentFormat.OpenXml.Packaging;
+using DocumentFormat.OpenXml.Spreadsheet;
+
+namespace Acceleratio.XCellKit
+{
+    public abstract class SpredsheetChart
+    {
+        public List<ChartModel> ChartData { get; protected set; }
+        public List<GanttData> GanttData { get; protected set; }
+        public ChartSettings Settings { get; set; }
+        internal ChartPropertiesSetup ChartPropertySetter { get; set; }
+
+        protected void ChartPropertyValuesFromSettings()
+        {
+            ChartPropertySetter.Title = Settings.Title ?? ChartPropertySetter.Title;
+            ChartPropertySetter.AxisX = Settings.AxisX ?? ChartPropertySetter.AxisX;
+            ChartPropertySetter.AxisXTitle = Settings.AxisXTitle ?? ChartPropertySetter.AxisXTitle;
+            ChartPropertySetter.AxisY = Settings.AxisY ?? ChartPropertySetter.AxisY;
+            ChartPropertySetter.AxisYTitle = Settings.AxisYTitle ?? ChartPropertySetter.AxisYTitle;
+            ChartPropertySetter.Height = Settings.Height ?? ChartPropertySetter.Height;
+            ChartPropertySetter.Legend = Settings.Legend ?? ChartPropertySetter.Legend;
+            ChartPropertySetter.SeriesColor = Settings.SeriesColor ?? ChartPropertySetter.SeriesColor;
+            ChartPropertySetter.Width = Settings.Width ?? ChartPropertySetter.Width;
+        }
+
+        protected SpredsheetChart(List<ChartModel> chartData, ChartSettings settings)
+        {
+            this.ChartData = chartData.Distinct().ToList();
+            this.Settings = settings;
+        }
+
+        protected SpredsheetChart(List<GanttData> ganttData, ChartSettings settings)
```

Comment:
It seems that GanttData is not used anywhere in this class, maybe transfer it to the GanttChart class? and then this constructor can only be protected SpredsheetChart(ChartSettings settings) ?